ml-agents/mlagents/trainers/learn.py

```diff
@@ -25,35 +25,194 @@
 from mlagents_envs.side_channel.engine_configuration_channel import EngineConfig
 
 
-class CommandLineOptions(NamedTuple):
-    debug: bool
-    seed: int
-    env_path: str
-    run_id: str
-    load_model: bool
-    train_model: bool
-    save_freq: int
-    keep_checkpoints: int
-    base_port: int
-    num_envs: int
-    curriculum_folder: Optional[str]
-    lesson: int
-    no_graphics: bool
-    multi_gpu: bool  # ?
-    trainer_config_path: str
-    sampler_file_path: Optional[str]
-    docker_target_name: Optional[str]
-    env_args: Optional[List[str]]
-    cpu: bool
-    width: int
-    height: int
-    quality_level: int
-    time_scale: float
-    target_frame_rate: int
+parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
```

Comment:
Note the parser is defined at the module level now -- this was needed in order to re-use the defaults in the `RunOptions` definition.

Reply:
Maybe move the setup to a _create_parser() function?

Reply:
Done